### PR TITLE
Fix some annoying warnings

### PR DESCRIPTION
**Pull request by Nekotekina**

### CODE REVIEW

**Comment by Nekotekina** [comment]
Go go brrrr.

File: rpcs3/Emu/Cell/Modules/cellImeJp.cpp [hunk 1]
```diff
@@ -43,10 +43,9 @@ bool ime_jp_manager::addChar(u16 c)
 	if (!c || cursor >= (CELL_IMEJP_STRING_MAXLENGTH - 1) || cursor > input_string.length())
 		return false;
 
-	std::u16string tmp;
-	tmp += c;
-	input_string.insert(cursor++, tmp);
-	cursor_end = cursor;
+	input_string += c; // resize
+	std::memmove(input_string.data() + cursor + 1, input_string.data() + cursor, sizeof(u16) * (input_string.size() - 1 - cursor));
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/cellImeJp.cpp:47]: I don't wanna test this but this really looks weird.
From a quick glance this entire change seems to change the logic.

File: rpcs3/Emu/Cell/Modules/cellAdec.cpp [hunk 2]
```diff
@@ -554,14 +530,7 @@ class AudioDecoder : public ppu_thread
...
 
-					last_frame = av_read_frame(fmt, &au) < 0;
-					if (last_frame)
-					{
-						//break;
-						av_free(au.data);
-						au.data = NULL;
-						au.size = 0;
-					}
+					av_read_frame(fmt, packet);
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/cellAdec.cpp:533]: Why is the result not checked?

File: rpcs3/Emu/Cell/Modules/cellAdec.cpp [hunk 3]
```diff
@@ -454,30 +454,8 @@ class AudioDecoder : public ppu_thread
...
-						}
-					}
-
-					~AVPacketHolder()
-					{
-						av_free(data);
-					}
-
-				} au(0);
+				AVPacket* packet = av_packet_alloc();
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/cellAdec.cpp:457]: Why is the object not checked for null?

File: rpcs3/Emu/Cell/Modules/cellSaveData.cpp [hunk 4]
```diff
@@ -148,7 +148,7 @@ int check_filename(std::string_view file_path, bool disallow_system_files, bool
 		return 70;
 	}
 
-	char name[CELL_SAVEDATA_FILENAME_SIZE - 3];
+	char name[CELL_SAVEDATA_FILENAME_SIZE + 3];
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/cellSaveData.cpp:151]: Why this change?

File: rpcs3/util/media_utils.cpp [hunk 5]
```diff
@@ -358,14 +358,14 @@ namespace utils
 			}
 
 			duration_ms = stream->duration / 1000;
-			
-			AVPacket packet{};
-			av_init_packet(&packet);
+
+			AVPacket* packet = av_packet_alloc();
```
Comment (Megamouse) [inline rpcs3/util/media_utils.cpp:362]: Why was this changed and why doesn't it check for null?

File: rpcs3/Emu/Cell/Modules/cellImeJp.cpp (hunk 1 above)
Comment (Nekotekina) [reply rpcs3/Emu/Cell/Modules/cellImeJp.cpp:47]: It should work like before, I did workaround compiler bug.

File: rpcs3/Emu/Cell/Modules/cellAdec.cpp (hunk 2 above)
Comment (Nekotekina) [reply rpcs3/Emu/Cell/Modules/cellAdec.cpp:533]: Previous code was destructor's job, so it shouldn't affect anything.

File: rpcs3/Emu/Cell/Modules/cellSaveData.cpp (hunk 4 above)
Comment (Nekotekina) [reply rpcs3/Emu/Cell/Modules/cellSaveData.cpp:151]: It's harmless but it fixes annoying compiler bug (or some real bug I couldn't figure out)

File: rpcs3/util/media_utils.cpp (hunk 5 above)
Comment (Nekotekina) [reply rpcs3/util/media_utils.cpp:362]: Idk, it's probably null only on out of memory, which itself is very serious problem which can't be worked around easily.

File: rpcs3/Emu/Cell/Modules/cellImeJp.cpp (hunk 1 above)
Comment (Megamouse) [reply rpcs3/Emu/Cell/Modules/cellImeJp.cpp:47]: It's completely broken now
